app.py

```diff
@@ -1,54 +1,79 @@
 from flask import Flask, render_template, request, redirect
+from flask_login import login_required, current_user
+from auth import init_auth, github_login
 from ViewModel import TodoListViewModel
 import board_utils
 import pymongo
 from bson.objectid import ObjectId
 import datetime
+import os
 
 
 def create_app():
     app = Flask(__name__)
+    if os.environ.get('LOGIN_DISABLED') is None or os.environ.get('LOGIN_DISABLED') != 'True':
+        app.config['LOGIN_DISABLED'] = False
+    else:
+        app.config['LOGIN_DISABLED'] = True
+    init_auth(app)
 
     mongo_manager = MongoConnectionManager()
 
     @app.route('/')
+    @login_required
     def index():
         print('getting all cards!!!')
         full_list = get_all_cards()
-        v_model = TodoListViewModel(board_utils.mapCardsToLocalRepresentation(full_list))
+        v_model = TodoListViewModel(board_utils.mapCardsToLocalRepresentation(full_list),
+                                    current_user_role_if_login_enabled())
         return render_template('index.html', v_model=v_model)
 
     @app.route('/add-list-item', methods=['POST'])
+    @login_required
     def addListItem():
-        print("Adding Item!")
-        card_title = request.form.get('new_card_textbox')
-        card = {
-            'name': card_title,
-            'dateLastActivity': datetime.datetime.now().isoformat()
-        }
-        devops_database = mongo_manager.get_database()
-        inserted_id = devops_database[board_utils.MONGO_LIST_TODO].insert_one(card).inserted_id
-        print('Created card with ID: ' + str(inserted_id))
+        if current_user_role_if_login_enabled() == "writer":
+            print("Adding Item!")
+            card_title = request.form.get('new_card_textbox')
+            card = {
+                'name': card_title,
+                'dateLastActivity': datetime.datetime.now().isoformat()
+            }
+            devops_database = mongo_manager.get_database()
+            inserted_id = devops_database[board_utils.MONGO_LIST_TODO].insert_one(card).inserted_id
+            print('Created card with ID: ' + str(inserted_id))
+        else:
+            print('User with role: ' + current_user_role_if_login_enabled() + ' does not have permission to add items')
         return redirect("/")
 
     @app.route('/completeditem', methods=['POST'])
+    @login_required
     def updateListItem():
-        print('Updating Item!')
-        print(request.form.get('id'))
-        card_id = request.form.get('id')
-        card_old_list = board_utils.MONGO_LIST_TODO
+        if current_user_role_if_login_enabled() == "writer":
```

Comment:
You could try writing a function decorator that performs this check to reduce duplicated code.